From 62d9fe5344ae74c4b614730ffcb1f68dbb561a41 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:47 -0400
Subject: [tools] perf: Add cpumode to struct hist_entry

Message-id: <1410181752-25631-283-git-send-email-jolsa@redhat.com>
Patchwork-id: 91918
O-Subject: [PATCH RHEL7.1 BZ1134356 282/307] perf tools: Add cpumode to struct hist_entry
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit 7365be55eee37ddb4f487263b4ba5bc8beb9638f
Author: Don Zickus <dzickus@redhat.com>
Date: Tue May 27 12:28:05 2014 -0400

description
===========
The next patch needs to sort on cpumode, so add it to hist_entry to be tracked.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/hist.c b/tools/perf/util/hist.c
index 5a0a4b2..d5f47a4 100644
--- a/tools/perf/util/hist.c
+++ b/tools/perf/util/hist.c
@@ -439,9 +439,10 @@ struct hist_entry *__hists__add_entry(struct hists *hists,
    .map = al->map,
    .sym = al->sym,
   },
-  .cpu = al->cpu,
-  .ip = al->addr,
-  .level = al->level,
+  .cpu  = al->cpu,
+  .cpumode = al->cpumode,
+  .ip  = al->addr,
+  .level  = al->level,
   .stat = {
    .nr_events = 1,
    .period = period,
diff --git a/tools/perf/util/sort.h b/tools/perf/util/sort.h
index 5bf0098..6de22f8 100644
--- a/tools/perf/util/sort.h
+++ b/tools/perf/util/sort.h
@@ -89,6 +89,7 @@ struct hist_entry {
  u64   ip;
  u64   transaction;
  s32   cpu;
+ u8   cpumode;
 
  struct hist_entry_diff diff;
 
-- 
1.7.1